lesson06/index.js

```diff
@@ -0,0 +1,28 @@
+const attributes = [
+    { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+    { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
+];
+
+const ul = document.getElementById("js-lists");
+
+const fragment = document.createDocumentFragment();
+
+const promise = new Promise(resolve => {
+    setTimeout(() => resolve(attributes), 3000);
+}).then(attributes => {
+    attributes.forEach(attribute => {
+        const li = document.createElement("li");
+        const anchor = document.createElement("a");
+        const img = document.createElement("img");
+    
+        anchor.textContent = attribute.text;
+        anchor.href = `/${attribute.to}`;
+        img.src = attribute.img;
+        img.alt = attribute.alt;
+    
+        li.appendChild(anchor).insertAdjacentElement("afterbegin", img);
+        fragment.appendChild(li);
+    });
+    ul.appendChild(fragment);
+});
+
```

Comment:
> After creating an instance with new Promise, is it hard to see the Arrow function overlapping?
```suggestion
const promise = new Promise(resolve => {
    setTimeout(() => resolve(attributes), 3000);
});

promise.then(attributes => {
 attributes.forEach(attribute => {
    const li = document.createElement("li");
    const anchor = document.createElement("a");
    const img = document.createElement("img");

    anchor.textContent = attribute.text;
    anchor.href = `/${attribute.to}`;
    img.src = attribute.img;
    img.alt = attribute.alt;

    li.appendChild(anchor).insertAdjacentElement("afterbegin", img);
    fragment.appendChild(li);
  });
  ul.appendChild(fragment);
});

```
How about dividing this part into different processes?
・Process to be executed by resolve after 3 seconds
・Process to create and add nodes.

I apologize if the answer is different from your question.
However, separating the processes will make the code easier to read and test.
This is something I learned in my review.

Reply:
wow!!! Separating the functions for each process made the code very readable!
I'll try to rewrite it.
Thank you for the review.